pandas/core/generic.py

```diff
@@ -2333,14 +2333,21 @@ def reindex_like(self, other, method=None, copy=True, limit=None,
 
         return self.reindex(**d)
 
-    def drop(self, labels, axis=0, level=None, inplace=False, errors='raise'):
+    def drop(self, labels=None, axis=0, index=None, columns=None, level=None,
+             inplace=False, errors='raise'):
         """
         Return new object with labels in requested axis removed.
 
         Parameters
         ----------
         labels : single label or list-like
+            Index or column labels to drop.
         axis : int or axis name
+            Whether to drop labels from the index (0 / 'index') or
+            columns (1 / 'columns').
+        index, columns : single label or list-like
```

Comment:
Perhaps add a `Notes` section saying that specifying both `labels` and `index` or `columns` will raise an error.